File: S3/DS/EXP4_BinarySearch/Binarysearch.c
#include<stdio.h>
void sort(int arr[],int n)
{
	int i,j,b,temp;
	for(i=0;i<n-1;i++)	
	{
		b=0;
		for(j=0;j<n-i-1;j++)
		{
			if(arr[j]>arr[j+1])
			{
				temp=arr[j];
				arr[j]=arr[j+1];
				arr[j+1]=temp;
				b=1;
			}
		}
		
		if(b==0)
			break;
	}
}
void search(int arr[],int n,int s)
{
	int fe=0,le=n,me,b=0;
	while(fe<=le)
	{
		me=fe+(le-fe)/2;
		if(arr[me]==s)
		{
			printf("The element is found at %d",me);
			b==1;
			break;
		}
		else if(arr[me]>s)
			le=me-1;
		else 
			fe=me+1;		
	}
		if (b==0)
			printf("The element is not found in the array");
		
}
		
int main()
{
	int a[100],n,i,s;
	printf("Enter the number of elements in the array");
	scanf("%d",&n);
	printf("Enter the array");
	for(i=0;i<n;i++)
	{
		scanf("%d",&a[i]);
	}
	printf("Enter the Search element");
	scanf("%d",&s);
	sort(a,n);
	search(a,n,s);
	printf("\n");
	return 0;
}
				

	

File: S3/OOPS/Lab/Cycle6/binarysearch/binarysearch.java
import java.util.*;
class search{
	void search(int a[],int key,int len)
	{
		int left=0,right=len;
		boolean b=false;
		while(left<right)
		{
			int mid=left+(right-left)/2;
			if(a[mid]==key)
			{
				System.out.println("The elements is found at "+mid);
				b=true;
				break;
			}
			else if(a[mid]>key)
			{
				right=mid-1;
			}
			else
			{
				left=left+1;
			}
		}
			if(b==false)
			{
				System.out.println("The element is not found at the array");
			}
	}
}
class main{
	public static void main(String[] args)
	{
		Scanner s=new Scanner(System.in);
		search obj=new search();
		int arr[]=new int[100];
		System.out.println("Enter the length of the array");
		int len=s.nextInt();
		System.out.println("Enter a sorted array");
		for(int i=0;i<len;i++)
		{
			arr[i]=s.nextInt();
		}
		System.out.println("Enter the search key");
		int key=s.nextInt();
		obj.search(arr,key,len);
		System.out.println();
	}
}
		
			
